### DISCUSSION TITLE
Growatt Thor EV Charger

### DISCUSSION THREAD

**Discussion by SvenPook** [lbbrhzn/ocpp#909] [Q&A] [answered]
I'm trying to configure the OCPP integration for use with my Growatt Thor EV Charger. I haven't been succesful. Can anyone tell me what the various settings should be?

The Growatt Thor is working correctly, tested and verified with the ShonePhone app and the Growatt Dashboard.

**Comment by GianQv** [comment, marked as answer]
Hi, in ShinePhone app open GroHome section -> Select the wallbox -> Ap Mode -> Server Settings.
Set here your ocpp server (in home assistant 'http://yourhassip:ocppserverport")
Password for Server settings '000000'
In home assistant ocpp integration put in wallbox id the serial number of the wallbox (ex 'WWJ0000XXXXXXXX')
Remember that the wallbox will not work with ShinePhone app if you change the ocpp server.

**Reply by OleMadsen1971** [reply]
I also have the Growatt Thor EV changer with 11KWh/16 amps.
To use the solar power i have to reduce the amps from 16 to the minimum, which is 6 apms (4,1KWh). I can do this today in the ShinePhone app.
But in Home Assistant, I'm only able to start/stop the charching in the OCPP integration.
Do you have any idea on how to set/change the amps during charching. So that I only charge when the sun i shining, and my solarpanels can at most only deliver about 7KWh (about 10 amps)

**Comment by SvenPook** [comment]
I’ve got a solution for that: numeric slider and a bunch of statements in a choose constructor. I’ll send it later today.Verstuurd vanaf mijn iPhoneOp 26 aug 2024 om 00:07 heeft OleMadsen1971 ***@***.***> het volgende geschreven:﻿
I also have the Growatt Thor EV changer with 11KWh/16 amps.
To use the solar power i have to reduce the amps from 16 to the minimum, which is 6 apms (4,1KWh). I can do this today in the ShinePhone app.
But in Home Assistant, I'm only able to start/stop the charching in the OCPP integration.
Do you have any idea on how to set/change the amps during charching. So that I only charge when the sun i shining, and my solarpanels can at most only deliver about 7KWh (about 10 amps)

**Comment by SvenPook** [comment]
this is the automation that I have. Note that the values under 6 do not work. 6 amps is the minimum that can be used with the charger. I take the input from a slider ( input_number.growatt_thor_max_charge_current ) and set a service shell command accordingly.

choose:

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

1.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 1

 - service: shell_command.thor_set_power_01

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

2.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 2

 - service: shell_command.thor_set_power_02

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

3.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 3

 - service: shell_command.thor_set_power_03

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

4.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 4

 - service: shell_command.thor_set_power_04

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

5.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 5

 - service: shell_command.thor_set_power_05

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

6.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 6

 - service: shell_command.thor_set_power_06

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

7.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 7

 - service: shell_command.thor_set_power_07

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

8.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 8

 - service: shell_command.thor_set_power_08

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

9.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 9

 - service: shell_command.thor_set_power_09

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

10.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 10

 - service: shell_command.thor_set_power_10

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

11.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 11

 - service: shell_command.thor_set_power_11

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

12.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 12

 - service: shell_command.thor_set_power_12

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

13.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 13

 - service: shell_command.thor_set_power_13

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

14.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 14

 - service: shell_command.thor_set_power_14

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

15.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 15

 - service: shell_command.thor_set_power_15

data: {}

 - conditions:

 - condition: template

value_template: >-

{{ (states("input_number.growatt_thor_max_charge_current") | float) ==

16.0 }}

sequence:

 - device_id: 4ad5d9619c3562b43fb904d760020361

domain: mobile_app

type: notify

message: val 16

 - service: shell_command.thor_set_power_16

data: {}

----------------------------------------

**Reply by OleMadsen1971** [reply]
I don't know where to put your script?
The slider in my HA does not change the 6A to 16A? It does not do anything?
And mine is named "charger_maximum_current"
Anf if I create a script within the Charger settings it creates this YAML, where I also can see the device ID is different from Yours.
But again, it does not set the "Max Output Current Set(6~32A):" setting on my Growatt charger, exactly the same as the slider, that does not change anything?
So I'm a bit lost? Mayby it's the initial setup of OCPP ?
-------------------------------------------------------
device_id: 491b7ce79f94d50eab89826f1b5feb19
domain: number
entity_id: e4df150c4277f17801cd176a25d5194a
type: set_value
value: 8

**Reply by SvenPook** [reply]
ok, sorry for the enormous long delay

in configuration.yaml place following. But adjust the ip address of your Thor unit. Mine is 192.168.5.52

# Growatt Thor
shell_command:
  thor_set_power_01: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=1.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_02: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=2.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_03: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=3.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_04: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=4.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_05: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=5.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_06: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=6.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_07: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=7.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_08: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=8.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_09: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=9.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_10: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=10.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_11: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=11.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_12: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=12.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_13: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=13.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_14: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=14.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_15: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=15.0' --insecure http://192.168.5.52:8080/config.cgi"
  thor_set_power_16: "curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.5.52:8080' -H 'Referer: http://192.168.5.52:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=16.0' --insecure http://192.168.5.52:8080/config.cgi"

**Reply by OleMadsen1971** [reply]
Thanks for the reply :-)

But where do you put the password for the Thor EV Charger (mine is default 12345678), or it might not be needed this way?
But the slider still does not work after editing configuration.yaml with your input.

If I execute the curl command from a terminal is is like this:
curl -X POST -H 'Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,/;q=0.8,application/signed-exchange;v=b3;q=0.7' -H 'Accept-Language: cs-CZ,cs;q=0.9,en;q=0.8' -H 'Cache-Control: max-age=0' -H 'Connection: keep-alive' -H 'Content-Type: application/x-www-form-urlencoded' -H 'Origin: http://192.168.1.133:8080' -H 'Referer: http://192.168.1.133:8080/' -H 'Upgrade-Insecure-Requests: 1' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36' --data-raw 'power=16.0' --insecure http://192.168.1.133:8080/config.cgi

Then it returns:
curl: (7) Failed to connect to 192.168.1.133 port 8080 after 0 ms: Could not connect to server

The firmware is: THOR_11AS-V2.2.16-20240902

**Reply by SvenPook** [reply]
I think you need to change some settings within your Thor.

Server URL(MaxLen 250):   ws://xxx.xxx.xxx.xxx:9000
use ip adrdress of your home assistant server

Firmware of my Thor:  THOR_11AS-V2.2.9-20221019
I don't know where I can find a new firmware. On the other hand, this one works fine for me

**Reply by OleMadsen1971** [reply]
Hi again
Thanks for your answer.

I have changed the IP on the charger to point to HA, otherwise you cannot stop/start the charger from HA.

Server URL(MaxLen 250): ws://<<my HA IP>>:9000

I have added your data with changed IP into configuration.yaml, and changed the IP from growatt to HA, and now i can stop/start the charge from HA, and see firmware, charge time, kwh charged and more. But the only thing missing is the change of 6A, 7A, 8A... until 16A, for me to use only solar power.

But the configuration.yaml cannot stand alone?
Do i have to add something changes to the settings in HA for your curl commands to work? Or how do they trigger?

Automations
No automatiseringer have been added using this enhed yet.
Scenes
No scener have been added using this enhed yet.
Scripts
No scripts have been added using this enhed yet.

<img width="697" height="790" alt="image" src="https://github.com/user-attachments/assets/436f2124-002e-463e-a000-9287737cc1ce" />

<img width="621" height="41" alt="image" src="https://github.com/user-attachments/assets/23aba63d-6385-4c20-b6ad-92f5c0b5c124" />

There is also a OCPP services.yaml. Have you made any changes there?

<img width="883" height="527" alt="image" src="https://github.com/user-attachments/assets/56c00440-c96f-40cd-a82e-deae6a36f3ed" />

**Reply by SvenPook** [reply]
That slider didn't work for me either, but.....I created an automation to solve that. The code is shown below:

alias: Growatt Thor max charge current
description: ""
mode: single
triggers:
  - entity_id:
      - input_number.growatt_thor_max_charge_current
    trigger: state
actions:
  - choose:
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 1.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 1
          - data: {}
            action: shell_command.thor_set_power_01
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 2.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 2
          - data: {}
            action: shell_command.thor_set_power_02
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 3.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 3
          - data: {}
            action: shell_command.thor_set_power_03
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 4.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 4
          - data: {}
            action: shell_command.thor_set_power_04
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 5.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 5
          - data: {}
            action: shell_command.thor_set_power_05
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 6.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 6
          - data: {}
            action: shell_command.thor_set_power_06
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 7.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 7
          - data: {}
            action: shell_command.thor_set_power_07
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 8.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 8
          - data: {}
            action: shell_command.thor_set_power_08
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 9.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 9
          - data: {}
            action: shell_command.thor_set_power_09
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 10.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 10
          - data: {}
            action: shell_command.thor_set_power_10
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 11.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 11
          - data: {}
            action: shell_command.thor_set_power_11
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 12.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 12
          - data: {}
            action: shell_command.thor_set_power_12
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 13.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 13
          - data: {}
            action: shell_command.thor_set_power_13
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 14.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 14
          - data: {}
            action: shell_command.thor_set_power_14
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 15.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 15
          - data: {}
            action: shell_command.thor_set_power_15
      - conditions:
          - condition: template
            value_template: >-
              {{ (states("input_number.growatt_thor_max_charge_current") |
              float) == 16.0 }}
        sequence:
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: val 16
          - data: {}
            action: shell_command.thor_set_power_16

**Reply by SvenPook** [reply]
Did you manage to get it working?

**Reply by OleMadsen1971** [reply]
No, not yet. When you copy/paste text in GitHub i guess it is formatted different. I have to figure out how to tab in this script?

Error when parsing YAML: can not read a block mapping entry; a multiline key may not be an implicit key (linje: 8, kolonne: 8)

And if I tab in it get this:
Fejl ved parsing af YAML: bad indentation of a mapping entry (linje: 8, kolonne: 14)

It does not like line 8:       trigger: state

I have tab it like this:
<img width="590" height="523" alt="image" src="https://github.com/user-attachments/assets/3feb558d-64e7-495d-b8e8-970cd1831b5a" />

Also? Should i change this number? 
device_id: 4ad5d9619c3562b43fb904d760020361

**Reply by OleMadsen1971** [reply]
It put the script into Google Gemini, and it changed it for me. Now i just have to change the device ID:
<img width="551" height="30" alt="image" src="https://github.com/user-attachments/assets/38e19f30-56da-49ed-8df6-31220db21480" />



alias: Growatt Thor max charge current
description: "Sets the Growatt Thor max charge current based on an input number."
mode: single
trigger:
  - platform: state
    entity_id: input_number.growatt_thor_max_charge_current
action:
  - choose:
      - conditions:
          - condition: template
            value_template: >-
              {{ (states('input_number.growatt_thor_max_charge_current') | float) >= 1.0 and 
                 (states('input_number.growatt_thor_max_charge_current') | float) <= 16.0 }}
        sequence:
          - service: shell_command.thor_set_power_{{ states('input_number.growatt_thor_max_charge_current') | int | string }}
          - device_id: 4ad5d9619c3562b43fb904d760020361
            domain: mobile_app
            type: notify
            message: "val {{ states('input_number.growatt_thor_max_charge_current') | int }}"
    default:
      - service: persistent_notification.create
        data:
          title: "Growatt Thor Max Charge Current Automation Error"
          message: "The input_number.growatt_thor_max_charge_current value is outside the valid range (1-16)."

**Reply by OleMadsen1971** [reply]
I got the device_id :-)
It is in HA on the charger in the link
config/devices/device/774a47a91c78b45f88328f03cc7cef2e

Now i have to figure out this part:
<img width="671" height="395" alt="image" src="https://github.com/user-attachments/assets/0c8fddb1-736b-4467-9fbf-c43486fcf40f" />

**Reply by SvenPook** [reply]
You can delete those 16 entries. I've had that for debugging purposes. It would send me a notification on my phone.

**Comment by stfvrg** [comment]
> Hi, in ShinePhone app open GroHome section -> Select the wallbox -> Ap Mode -> Server Settings.
> 
> Set here your ocpp server (in home assistant 'http://yourhassip:ocppserverport")
> 
> Password for Server settings '000000'
> 
> In home assistant ocpp integration put in wallbox id the serial number of the wallbox (ex 'WWJ0000XXXXXXXX')
> 
> Remember that the wallbox will not work with ShinePhone app if you change the ocpp server.

It has to be ws://yourHAip:ocppserverport
It needs a web socket connection instead of an http connection.

**Reply by hnktnk** [reply]
I think I named it http and now I'm locked out from my Thor 👯 
Gonna try and reset the thing... somehow.

**Reply by OleMadsen1971** [reply]
Just type the string that connects to Growatt again, and your fine:
The entry at "Server URL(MaxLen 250)": ws://evcharge.growatt.com:80/ocpp/ws

**Reply by hnktnk** [reply]
thanks, now I need to get the webui back working.
